client/db/src/record_stats_state.rs

```diff
@@ -73,10 +73,43 @@ impl<S: StateBackend<HashFor<B>>, B: BlockT> RecordStatsState<S, B> {
 	}
 }
 
+pub struct RawIter<S, B>
+where
+	S: StateBackend<HashFor<B>>,
+	B: BlockT,
+{
+	inner: <S as StateBackend<HashFor<B>>>::RawIter,
+}
+
+impl<S, B> StorageIterator<HashFor<B>> for RawIter<S, B>
+where
+	S: StateBackend<HashFor<B>>,
+	B: BlockT,
+{
+	type Backend = RecordStatsState<S, B>;
+	type Error = S::Error;
+
+	fn next_key(&mut self, backend: &Self::Backend) -> Option<Result<StorageKey, Self::Error>> {
```

Comment:
Not for this PR, but could be good to record state (at least adding it looks straight forward).